sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala

```diff
@@ -50,7 +50,7 @@ case class ParquetScanBuilder(
     val pushDownInFilterThreshold = sqlConf.parquetFilterPushDownInFilterThreshold
     val isCaseSensitive = sqlConf.caseSensitiveAnalysis
     val parquetSchema =
-      new SparkToParquetSchemaConverter(sparkSession.sessionState.conf).convert(schema)
```

Comment:
What if the partition column is in file? I guess this might be why we push down partition filter.

Reply:
This is a good question, but it seems that the filter pushed down in DataSource V1 does not contain the filter related to partition columns too. 

The `dataFilters` use to construct `FileSourceScanExec` and pass to `ParquetFileFormat.buildReaderWithPartitionValues` to build `pushed` filters also filter out partition filters, am I right?

https://github.com/apache/spark/blob/e4d1c10760800563d2a30410b46e5b0cd2671c4d/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala#L184-L193

Reply:
Maybe another way is to refer to  parquet `FileMetaData#getSchema` to determine which filter should be pushed down, but this needs to be done once for each file.

Reply:
Maybe this is a correct fix. I have a pr to fix partition column is in file.

Reply:
Maybe we should use `readDataSchema()` instead of `dataSchema`

Reply:
Address b9f8eb2 use `readDataSchema()` instead of `dataSchema`